downloader/downloader.go

```diff
@@ -115,12 +115,11 @@ func getVideoAudioFormats(v *youtube.Video, quality string, mimetype string) (*y
 	var videoFormat, audioFormat *youtube.Format
 	var videoFormats, audioFormats youtube.FormatList
 
-	// Default mime-type to mp4
-	if mimetype == "" {
-		mimetype = "mp4"
+	formats := v.Formats
+	if mimetype != "" {
+		formats = formats.Type(mimetype)
```

Comment:
@Julian-Chu I removed the default choice here to avoid double defaults, since flag already has default value.

Reply:
I  suggest to return  an error, when mimetype is empty string, to keep the responsibility and code of mimetype handling  by caller (e.q. cmd) 

ps: don't forget to have a simple unit test for the empty mimetype error :beer:

Reply:
Since getVideoAudioFormats() can operate with either quality or mimetype or even none of these paramters, I think 🤔it's ok to leave like this? do you think mimetype should be required? if yes, should quality too? The enforced mp4 mimetype was to reflect the previous behavior https://github.com/kkdai/youtube/commit/ab294fc658645bd524e836a5b5fe2bde1849c1e3#diff-0560d473838290dc1f7cb1ae6dce590e1e202d9c5dea4bc077c4e4cc98541f93R62-R77

Current flow:
- if mimetype is set, filter all formats (video / audio) by mimetype 
- if quality is set, filter video formats (already filtered by mimetype if mimetype is given) -- since quality is dedicated to video
- last step, sort formats with some rules to favor best quality then take the first format in video & audio lists to return to caller (https://github.com/kkdai/youtube/blob/382f2ff39a6efd3d5bc71d43a1bad3512bc58750/format_list.go#L78)

--

Examples with current code with optional mimetype in getVideoAudioFormats

Without mimetype, video/webm and audio/mp4 are chosen because the precedence is `av01, vp9, avc1` for video and `mp4, opus` for audio which is buggy (I've just found out) when webm or mp4 is selected for a stream the other one must follow☺
```
-v download --quality="hd1080" --mimetype="" ADFgbePK7gw

Video 'Bells for Her' - Quality '1080p' - Video Codec 'video/webm; codecs="vp9"' - Audio Codec 'audio/mp4; codecs="mp4a.40.2"'
```

default mimetype is mp4
```
-v download --quality="hd1080" ADFgbePK7gw

Video 'Bells for Her' - Quality '1080p' - Video Codec 'video/mp4; codecs="avc1.640020"' - Audio Codec 'audio/mp4; codecs="mp4a.40.2"'
```

mimetype set to webm
````
-v download --quality="hd1080" --mimetype="webm" ADFgbePK7gw

Video 'Bells for Her' - Quality '1080p' - Video Codec 'video/webm; codecs="vp9"' - Audio Codec 'audio/webm; codecs="opus"'
```

Reply:
thanks for the explanation, let's keep it  in this PR.